charts/cluster-api-runtime-extensions-nutanix/templates/cluster-autoscaler/manifests/helm-addon-installation.yaml

```
@@ -9,7 +9,7 @@ metadata:
 data:
   values.yaml: |-
     ---
-    fullnameOverride: "cluster-autoscaler-{{ `{{ .Cluster.Name }}` }}"
+    fullnameOverride: "cluster-autoscaler-{{ `{{ md5sum (printf "%s/%s" .Cluster.Namespace .Cluster.Name) }}` }}"
```

Do we want to add a label here? I tested this locally, the same k:v is already applied to the Cluster object so there is no risk of applying labels that are too long.
```suggestion
    fullnameOverride: "cluster-autoscaler-{{ `{{ md5sum (printf "%s/%s" .Cluster.Namespace .Cluster.Name) }}` }}"
    
    additionalLabels:
      cluster.x-k8s.io/cluster-name: "{{ `{{ .Cluster.Name }}` }}"
```

---

Good idea. I'll add cluster namespace too when I get a chance

---

Would this data become stale if the cluster is "moved" (`clusterctl move`) to a different namespace?

---

Now handled by adding a UUID annotation (our own, not `metadata.uid`) via defaulting webhook and enforced immutability via a validating webhook. This is then moved with the cluster so this UUID will remain stable regardless of moving the cluster anywhere.